src/main/java/org/primefaces/component/export/CSVExporter.java

```diff
@@ -184,20 +198,20 @@ protected void addColumnValue(StringBuilder builder, UIComponent component) thro
     }
 
     protected void addColumnValue(StringBuilder builder, String value) throws IOException {
-        value = (value == null) ? "" : value.replaceAll("\"", "\"\"");
+        value = (value == null) ? "" : value.replace(csvOptions.getQuoteString(), csvOptions.getDoubleQuoteString());
 
```

Comment:
Just curious why you chose replace instead of the original replaceAll to find and escape all the quotes in the string?

Reply:
fantastic answer.  I support this 100%.

Reply:
Interesting, I had the exact opposite result running the benchmark snippet above.

```java
replaceAll took 56330 ms
replace took 127743 ms
```
```
Linux 4.4.0-128-generic
openjdk version "1.8.0_171"
```


To be fair, you can paste the snippet to some online Java IDE such as https://www.compilejava.net/
(There is a 5 seconds time limit so you can just reduce the number of repeats :eyes: )
```java
public class Test {    
    public static void main(String[] args) {
		final String str = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789\"9876543210ZYXWVUTSRQPONMLKJIHGFEDCBA";
		final String quote = "\"", doubleQuote = "\"\"";
		final int repeats = 1_000_000;
		
		long start = System.currentTimeMillis();
		for (int i = 0; i < repeats; i++) {
			str.replaceAll(quote, doubleQuote);
		}
		System.out.println(String.format("replaceAll took %d ms", System.currentTimeMillis() - start));

		start = System.currentTimeMillis();
		for (int i = 0; i < repeats; i++) {
			str.replace(quote, doubleQuote);
		}
		System.out.println(String.format("replace took %d ms", System.currentTimeMillis() - start));
	} 
}
```

![snapshot1](https://user-images.githubusercontent.com/10467831/42109240-9b65d0f8-7ba2-11e8-9445-0027b3af7442.png)

Reply:
@jxmai Probably there are differences in JVM/OS/CPU setup.

My setup: Windows 8.1, Intel Core i7-3770 @ 3.4GHz (4 cores), 100e6 repetitions

**Oracle JDK 9.0.4**
```
replaceAll took 64815 ms
replace took 7111 ms
```

**OpenJDK 10.01**
```
replaceAll took 71586 ms
replace took 8083 ms
```

Reply:
**OpenJDK 9, Debian, Intel Xeon X5660 @ 2,8GHz (4 cores), 100e6 repetitions**
```
replaceAll took 52079 ms
replace took 12339 ms
```

Reply:
Solution of the mysterious behavior: @jxmai was using JDK 8 whereas @cnsgithub was using JDK 9/10.

**JDK 8**
```java
    public String replace(CharSequence target, CharSequence replacement) {
        return Pattern.compile(target.toString(), Pattern.LITERAL).matcher(
                this).replaceAll(Matcher.quoteReplacement(replacement.toString()));
    }
```

**JDK 9/10**
```java
    public String replace(CharSequence target, CharSequence replacement) {
        String tgtStr = target.toString();
        String replStr = replacement.toString();
        int j = indexOf(tgtStr);
        if (j < 0) {
            return this;
        }
        int tgtLen = tgtStr.length();
        int tgtLen1 = Math.max(tgtLen, 1);
        int thisLen = length();

        int newLenHint = thisLen - tgtLen + replStr.length();
        if (newLenHint < 0) {
            throw new OutOfMemoryError();
        }
        StringBuilder sb = new StringBuilder(newLenHint);
        int i = 0;
        do {
            sb.append(this, i, j).append(replStr);
            i = j + tgtLen;
        } while (j < thisLen && (j = indexOf(tgtStr, j + tgtLen1)) > 0);
        return sb.append(this, i, thisLen).toString();
    }
```

**Conclusion**
JDK ≥ 9 users profit from using `String.replace`. For JDK < 9 it should be irrelevant if we use `String.replace` or `String.replaceAll`.

Reply:
Yep I am using JDK8 also so that makes perfect sense.    Good to see they finally offered an optimized version of replace in JDK 9+!